worker/src/RTC/Transport.cpp

```diff
@@ -1726,7 +1727,9 @@ namespace RTC
 			default:;
 		}
 
-		delete packet;
+		// Media forwarded, the packet lifetime is already handled.
+		if (result != RTC::Producer::ReceiveRtpPacketResult::MEDIA_FORWARDED)
+			delete packet;
```

Comment:
This is super fragile, let's try to avoid such things when de-allocation of something depends on return value somewhere in completely different place.

Reply:
Handled in the last commit. Sorry @nazar-pc, indeed even though I marked it as PR I already told @ibc to wait for review until a missing commit was pushed..., I forgot to label this as draft again..